### PR TITLE
Signed requests snippet

### PR DESCRIPTION
**Pull request by nodebotanist**

### CODE REVIEW

File: content/templates/snippets/signed_request.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+title: 'Signed Request/Response'
+hidden: false
```
Comment (victoriabernard92) [inline content/templates/snippets/signed_request.md]:
```suggestion
hidden: true
```

File: data/snippet/signed_request.toml [hunk 2]
```diff
@@ -0,0 +1,49 @@
+id = "signed_request"
+name = "Signed Request/Response"
+description = "Check signatures of requestst and sign responses with a private key"
+code = """
+const crypto = require('crypto')
```
Comment (victoriabernard92) [inline data/snippet/signed_request.toml]: What do you think of adding a tag to the snippets
"ESM" with hover over "Requires ES Modules, so cannot view in previewer without webpack" and then in the demo link we can have the unminifed webpacked snippet code with the important code on top

Without something specifying this requires webpack it technically should be a boilerplate instead, but I actually prefer as a snippet also. Just need a way to communicate the difference

File: data/snippet/signed_request.toml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+async function signResponse(responseBody, response) {
+  // create signature
+  const signature = await createHexSignature(responseBody)
+  response.headers.set('signature', signature)
+  //add header with signature
+  return response
+}
+
+/**
+ * Fetch and log a request
```
Comment (victoriabernard92) [inline data/snippet/signed_request.toml]: Put actually comment of handleRequest
```suggestion
 * does something..
```

File: data/snippet/signed_request.toml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+  let hmac = crypto.createHmac('sha256', secret)
+  hmac.update(requestBody)
+  return hmac.digest('hex')
+}
+
+async function checkSignature(request) {
+  // hash request with secret key
+  let expectedSignature = await createHexSignature(await request.text())
+  let actualSignature = await request.headers.get('signature')
+
```
Comment (victoriabernard92) [inline data/snippet/signed_request.toml]: can't have \n\n. Hugo won't compile. I suggest running `npm run start-hugo` before submitted to check out what it looks like

File: data/snippet/signed_request.toml [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  // check that hash matches signature
+  return expectedSignature == actualSignature
```
Comment (victoriabernard92) [inline data/snippet/signed_request.toml]:
```suggestion
  return expectedSignature === actualSignature
```

File: data/snippet/signed_request.toml [hunk 6]
```diff
@@ -0,0 +1,49 @@
+id = "signed_request"
+name = "Signed Request/Response"
+description = "Check signatures of requestst and sign responses with a private key"
```
Comment (victoriabernard92) [inline data/snippet/signed_request.toml]:
```suggestion
description = "Check signatures of requests and sign responses with a private key"
```

File: data/snippet/signed_request.toml (hunk 2 above)
Comment (victoriabernard92) [reply data/snippet/signed_request.toml]: Note just curious of your opinion I can actually implement it if you like

Comment (nodebotanist) [reply data/snippet/signed_request.toml]: That sounds good to me.

**Pull request merged**